### ISSUE TITLE
Query Favorite and Placeholders incorrectly replaced

### ISSUE THREAD

**Issue by gms8994** [Sequel-Ace/Sequel-Ace#380]
- Sequel Ace Version: 2.1.8
- macOS Version: N/A
- MySQL Version: N/A

**Description**
Query favorites with `$` in them attempt to be replaced by variables no matter what (including when inside quotes).

**Steps To Reproduce**
1. Create a query favorite that uses a bcrypt generated string (`$2y$12$nZKuPAtDfXDuPeqp0hVnEukUKIUbTUUiE4JlIVZIR.RtAFwbW1Uue` from https://bcrypt-generator.com)
2. Select the favorite
3. Note that the value shown in the query window is `y$nZKuPAtDfXDuPeqp0hVnEukUKIUbTUUiE4JlIVZIR.RtAFwbW1Uue`.

**Expected Behaviour**
The value should be provided as is.

**Is Issue Present in [Latest Beta](https://github.com/Sequel-Ace/Sequel-Ace/releases)?**
Likely, but currently unsure

**Additional Context**
I understand about query placeholders, and it looks like only one ("Mirrored Tab Snippet") uses a value that does not include braces (`{` and `}`) - and even that seems to only ever use `$1`; perhaps something can be improved there? Or possibly something that suggests that "I don't want placeholders" in this query"? I'm not sure the best solution to the issue, unfortunately

**Comment by Jason-Morcos** [comment]
Probably this can be fixed by tweaking the Regex, something I am very very very bad at.

**Comment by gms8994** [comment]
@Jason-Morcos I'm pretty okay with regex; if you can show me where it lives, I might be able to assist? There's still the issue about the `$1` for Mirrored Tab Snippet, but having that as an "expected" edge case might be okay?

**Comment by Jason-Morcos** [comment]
> @Jason-Morcos I'm pretty okay with regex; if you can show me where it lives, I might be able to assist? There's still the issue about the `$1` for Mirrored Tab Snippet, but having that as an "expected" edge case might be okay?

I believe changes will need to be made to the `insertAsSnippet` method of `SPTextView.m`. Additionally, I'm not totally opposed to changing the Mirrored Tab Snippet as long as we clearly notify users!

**Comment by gms8994** [comment]
@Jason-Morcos I just looked at the code in SPTextView.m, and it looks like the regex is checking for an escaped $ (ie `\$`) in the snippet; maybe we could get away with some messaging in the favorite editor whenever an unescaped `$` is present? For example, if I paste in the above hash, it could provide a tooltip (or even just a message below the editor section) that suggests that if you want to use `$` in this way, you might consider escaping it? That way people can continue to use the snippets as is, but there's a warning to the user.

**Comment by Jason-Morcos** [comment]
I think that makes sense, @gms8994! I don't love the way it all works, but honestly can't really think of something much better. We could recommend $'s are escaped like you describe and potentially tighten the regex to also only recognize placeholders that are ${ or $1 so that an unescaped $ would only even become a factor for a mirrored tab character? Also we could change the mirrored tab snippet which would perhaps let us make the regex even tighter.

**Comment by luis-** [comment]
Hey @Jason-Morcos,

Came across this one from #1894. 
What if, to fix, we skip replacing mirrors (of form `$#`) only if "control" placeholder exist (of form`${#:...}` ?.

So from Original post, `$2y$12$nZKuPAtDfXDuPeqp0hVnEukUKIUbTUUiE4JlIVZIR.RtAFwbW1Uue` the `$2` and `$12` would be left in because no placeholders of form`${#:...}` exist.

I tried a quick fix where I only apply mirrors if \*any* control snippets exists (just based on counter in code being > -1) and that seems to let `$2y$12$nZKuPAtDfXDuPeqp0hVnEukUKIUbTUUiE4JlIVZIR.RtAFwbW1Uue` snippet to be inserted as is. Note, this quick fix isn't robust enough, preferrably I would modify insertion code to only mirror if corresponding control exists i.e. only mirror `$1` if `${1:...}` exists, only mirror `$2` if `${2:...}` exists, etc. but wanted to get your thoughts before diving into making the changes.

**Comment by MacGritsch** [comment]
Didnt even know of this placeholder-system in the favorites so I would support the change to disable that functionality of no variables (or however they are called) are defined.

**Comment by Jason-Morcos** [comment]
@luis- That seems to make sense! That would at least significantly limit the impact without hindering any functionality, correct?

**Comment by luis-** [comment]
Yep that's what I was aiming for. I can't think of a case where not replacing unused mirrors would be a serious issue for someone, worse case they would just need to edit their saved queries to remove unused mirrors.

I'll give fixing this a shot later in the week when my work-week winds down.
